project-addons/crm_claim_rma_custom/crm_claim.py

```diff
@@ -61,6 +68,17 @@ def onchange_claim_type(self):
             return {'domain': {'partner_id': [('supplier', '=', True),
                                               ('is_company', '=', True)]}}
 
+    @api.multi
+    def write(self, vals):
+        if 'stage_id'in vals and vals['stage_id'] == 3:
+            for line in self.claim_line_ids:
+                line_state = line.substate_id.string_id
+                if line_state not in self.check_states:
```

Comment:
Sí, no se porque me sonaba que lo que se quería hacer aquí es mostrar el error al pasar a un sólo subestado, posiblemente me lo confunda con la etapa, de todas formas, estás haciendo un "not in" de 6 elementos cuando sólo tienes 10 sub-estados, un "in" sería más corto y si se crean sub-eatados nuevos no entrarían en este código que de la otra forma sí lo harían.

Aprovechando veo que también en el if superior comparas el valor de stage_id con el id 3, este id también tiene un id externo, ya que las etapas se crean desde un xml igual que los subestados, lo puedes conocer de la misma forma.